addons/xivbar/main.lua

```diff
@@ -0,0 +1,231 @@
+            if new_hp_width == 0 then
+                hp_foreground:hide()
+            end
+            hp_update = false
+        end
+    end
+
+    hp_text:clear()
+    hp_text:append('' .. info.vitals.hp)
+end
+
+function update_mp()
+    local info         = windower.ffxi.get_player()
+    local old_mp_width = mp_foreground:width()
+    local new_mp_width = math.floor((info.vitals.mpp / 100) * bar_width)
+
+    if new_mp_width ~= nil and new_mp_width >= 0 then
+        if old_mp_width < new_mp_width then
+            local x = old_mp_width + math.ceil(((new_mp_width - old_mp_width) * 0.1))
+
+            if x > bar_width then
+                x = bar_width
+            end
+
+            mp_foreground:show()
+            mp_foreground:size(x, total_height)
+        elseif old_mp_width > new_mp_width then
+            local x = old_mp_width - math.ceil(((old_mp_width - new_mp_width) * 0.1))
+
+            if x < 0 then
+                x = 0
+            end
+
+            mp_foreground:show()
+            mp_foreground:size(x, total_height)
+        elseif old_mp_width == new_mp_width then
+            if new_mp_width == 0 then
+                mp_foreground:hide()
+            end
+            mp_update = false
+        end
+    end
+
+    mp_text:clear()
+    mp_text:append('' .. info.vitals.mp)
+end
+
+function update_tp()
+    local info         = windower.ffxi.get_player()
+    local old_tp_width = tp_foreground:width()
+    local new_tp_width = bar_width
+
+    if info.vitals.tp < 1000 then
+        new_tp_width = math.floor((info.vitals.tp / 1000) * bar_width)
+    end
+
+    if new_tp_width ~= nil and new_tp_width >= 0 then
+        if old_tp_width < new_tp_width then
+            local x = old_tp_width + math.ceil(((new_tp_width - old_tp_width) * 0.1))
+
+            if x > bar_width then
+                x = bar_width
+            end
+
+            tp_foreground:show()
+            tp_foreground:size(x, total_height)
+        elseif old_tp_width > new_tp_width then
+            local x = old_tp_width - math.ceil(((old_tp_width - new_tp_width) * 0.1))
+
+            if x < 0 then
+                x = 0
+            end
+
+            tp_foreground:show()
+            tp_foreground:size(x, total_height)
+        elseif old_tp_width == new_tp_width then
+            if new_tp_width == 0 then
+                tp_foreground:hide()
+            end
+            tp_update = false
+        end
+    end
+
+    tp_text:clear()
+    tp_text:append('' .. info.vitals.tp)
+end
+
+function hide()
+    background:hide()
+    hp_foreground:hide()
+    hp_text:hide()
+    mp_foreground:hide()
+    mp_text:hide()
+    tp_foreground:hide()
+    tp_text:hide()
+    ready = false
+end
+
+function show()
+    background:show()
+    hp_foreground:show()
+    hp_text:show()
+    mp_foreground:show()
+    mp_text:show()
+    tp_foreground:show()
+    tp_text:show()
+    ready = true
+end
+
+function initialize()
```

Comment:
This function is called every time you log in, but really only needs to be called once.